library/src/main/java/com/parse/internal/signpost/basic/DefaultOAuthProvider.java

```diff
@@ -38,7 +38,7 @@ public DefaultOAuthProvider(String requestTokenEndpointUrl, String accessTokenEn
 
     protected HttpRequest createRequest(String endpointUrl) throws MalformedURLException,
             IOException {
-        HttpURLConnection connection = (HttpURLConnection) new URL(endpointUrl).openConnection();
+        HttpURLConnection connection = HttpURLConnectionProvider.getInstance().open(new URL(endpointUrl));
```

Comment:
Let's pass in a `HttpURLConnectionProvider` instance through the constructor to match the `CommonsHttpOAuthProvider`.